src/components/container/SurveyTaskList/SurveyTaskList.tsx

```diff
@@ -28,19 +29,30 @@ export type SurveyTaskListListPreviewState = "IncompleteTasks" | "CompleteTasks"
 
 export default function (props: SurveyTaskListProps) {
 	const [loading, setLoading] = useState(true);
-	const [tasks, setTasks] = useState<SurveyTask[] | null>(null);
+	const [tasks, setTasksInner] = useState<SurveyTask[] | null>(null);
+	const [activeSurveys, setActiveSurveys] = useState<string[]>([]);
 	const context = useContext(LayoutContext);
 
-	useEffect(() => {
-		initialize()
-		MyDataHelps.on("applicationDidBecomeVisible", initialize);
-		return () => {
-			MyDataHelps.off("applicationDidBecomeVisible", initialize);
+	const setTasks = (tasks: SurveyTask[]): void => {
+		setTasksInner(tasks);
+		setActiveSurveys([]);
+	};
+
+	useInitializeView(initialize, [], [props.previewState]);
```

Comment:
Oh wow, I totally missed that you added useInitializeView.  I actually added a very similar hook which I should probably get rid of, since it is just less good: https://github.com/CareEvolution/MyDataHelpsUI/blob/main/src/helpers/useMyDataHelps.ts